src/main/msp/msp_build_info.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdint.h>
+
+#include "platform.h"
+
+#include "common/streambuf.h"
+
+#include "msp/msp_build_info.h"
+
+void sbufWriteBuildInfoFlags(sbuf_t *dst) 
+{
+    uint8_t options = 0;
+#ifdef USE_SERIALRX_CRSF
+    options |= MSP_BUILD_OPTION_FLAG_SERIALRX_CRSF;
+#endif
+#ifdef USE_SERIALRX_FPORT
+    options |= MSP_BUILD_OPTION_FLAG_SERIALRX_FPORT;
+#endif
+#ifdef USE_SERIALRX_GHST
+    options |= MSP_BUILD_OPTION_FLAG_SERIALRX_GHST;
+#endif
+#ifdef USE_SERIALRX_IBUS
+    options |= MSP_BUILD_OPTION_FLAG_SERIALRX_IBUS;
+#endif
+#ifdef USE_SERIALRX_JETIEXBUS
+    options |= MSP_BUILD_OPTION_FLAG_SERIALRX_JETIEXBUS;
+#endif
+#ifdef USE_RX_PPM
+    options |= MSP_BUILD_OPTION_FLAG_RX_PPM;
+#endif
+#ifdef USE_SERIALRX_SBUS
+    options |= MSP_BUILD_OPTION_FLAG_SERIALRX_SBUS;
+#endif
+#ifdef USE_SERIALRX_SPEKTRUM
+    options |= MSP_BUILD_OPTION_FLAG_SERIALRX_SPEKTRUM;
+#endif
+    sbufWriteU8(dst, options); // byte 1
```

Comment:
Perhaps we define `radioProtocol`, `radioTelemetry` and `otherOptions` in different 16 bit values to not have to mix them up.

Reply:
I was thinking about logical grouping too, as the current order of the bits simply reflects the order of items in the JSON response (C code was generated using a Python script). Anyway, in the future it will be only possible to append new flags at the end of the MSP payload to keep backwards compatibility. In this sense the order of things becomes less significant because the position will only mean "it was added at a later point in time". At this point I'm not sure about how much we should care about the placement of bits. I think users will just learn that option Z is set at bit Y in byte X and that's it.

Reply:
Maybe let the client ask for availability of options and they will be returned in the same order in the response? I cannot think of any better solution to this problem right now